bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/config/BigtableVeneerSettingsFactory.java

```diff
@@ -97,6 +99,15 @@ public static BigtableDataSettings createBigtableDataSettings(
           buildChannelProvider(dataSettingStub.getEndpoint(), options));
     }
 
+    if (options.getTracingCookie() != null) {
+      Map<String, String> headers =
+          ImmutableMap.<String, String>builder()
+              .putAll(dataSettingStub.getHeaderProvider().getHeaders())
+              .put("cookie", options.getTracingCookie())
+              .build();
+      dataSettingStub.setHeaderProvider(FixedHeaderProvider.create(headers));
```

Comment:
I think this will clobber the existing header provider that is setting the user agent

Reply:
Moved this to `buildHeaderProvider`. I think also `TestHeaders` is testing tracing header with user agent header?